client/src/App.tsx

```diff
@@ -1382,16 +1391,21 @@ function App() {
 					isMovingBoardObject={!!movingBoardItem}
 				/>
 			)}
-
 			<Modal
-				onClose={() => setIsModalOpen(false)}
+				onClose={() => setModalState('none')}
 				className="modal-container"
-				open={isModalOpen}
+				open={modalState === 'new-room' || modalState === 'enter-room'}
```

Comment:
if you use `null` instead of 'none', you can do a shortcut for open, check if the value exists or not (by turning it into a boolean through `!!`)

`open={!!modalState}`